Problems with the python interface on MAC:

1. You are getting the error message "Fatal Python error: PyThreadState_Get: no current thread"
when you do "python call_correlation_functions.py"

This is usually caused by a python library mis-match between linking time and run-time, i.e., 
the python shared extension (_countpairs.so in the python_bindings/ subdir) was built with a
python lib version that is not the first one in your search path. The way I resolved it on 
my MAC was by adding the path: "python-config --prefix"/lib to the dynamic library 
path environment variable. 

Typically, this happens when the output of "otool -L _countpairs.so" does not show 
a full path to libpythonX.X.dylib. 

In my case, I use the anaconda distribution for python at link-time but I pick 
up the macports install at run-time. The exact command to fix the situation is:

Option 1.
----------
Change the rpath in the shared C extension library: 

"install_name_tool -change libpythonX.X.dylib `python-config --prefix`/lib/libpythonX.X.dylib _countpairs.so"

"otool -L _countpairs.so" should show the full path to the libpythonX.X.dylib file. 

Option 2. 
----------
Add to the environment variable - but this is not fool-proof. 
"export DYLD_FALLBACK_LIBRARY_PATH=`python-config --prefix`/lib:$DYLD_FALLBACK_LIBRARY_PATH"

Option 3.
----------
If that does not work, try creating a symbolic link in the python_bindings directory:

"ln -s `python-config --prefix`/lib/libpythonX.X.dylib" 

(where I have omitted the sym-link name, defaults to the actual filename.) 

However, if you go this sym-link route, then the sym-link *has* to be in the directory
where the python code is situated - I do not know of any work-arounds (short of 
creating the sym-link in a directory that's in the dynamic library path). 

